src/ml_tooling/baseclass.py

```diff
@@ -326,83 +335,85 @@ def default_metric(self):
                 else self.config.REGRESSION_METRIC)
 
     @default_metric.setter
     def default_metric(self, metric):
-        if self.model._estimator_type == 'classifier':
+        if self.estimator._estimator_type == 'classifier':
             self.config.CLASSIFIER_METRIC = metric
         else:
             self.config.REGRESSION_METRIC = metric
 
     @classmethod
     def test_models(cls,
-                    models: Sequence,
+                    estimators: Sequence,
                     metric: Optional[str] = None,
                     cv: Union[int, bool] = False,
-                    log_dir: str = None) -> Tuple['BaseClassModel', ResultGroup]:
+                    log_dir: str = None) -> Tuple['ModelData', ResultGroup]:
         """
-        Trains each model passed and returns a sorted list of results
+        Trains each estimator passed and returns a sorted list of results
 
         Parameters
         ----------
-        models: Listlike
+        estimators: Sequence
             List of models to train
 
         metric: str, optional
-            Metric to use in scoring of model
+            Metric to use in scoring of estimators
 
         cv: int, bool
             Whether or not to use cross-validation. If an int is passed, use that many folds
 
         log_dir: str, optional
-            Where to store logged models. If None, don't log
+            Where to store logged estimators. If None, don't log
 
         Returns
         -------
         List of Result objects
         """
         results = []
 
-        for i, model in enumerate(models, start=1):
-            logger.info(f"Training model {i}/{len(models)}: {_get_model_name(model)}")
-            challenger_model = cls(model)
-            result = challenger_model.score_model(metric=metric, cv=cv)
+        for i, estimator in enumerate(estimators, start=1):
+            logger.info(f"Training estimator {i}/{len(estimators)}: "
+                        f"{_get_estimator_name(estimator)}")
+            challenger_model = cls(estimator)
+            result = challenger_model.score_estimator(metric=metric, cv=cv)
             results.append(result)
             if log_dir:
-                result.log_model(log_dir)
+                result.log_estimator(log_dir)
 
         results.sort(reverse=True)
-        best_model = results[0].model
+        best_estimator = results[0].estimator
         logger.info(
-            f"Best model: {results[0].model_name} - {results[0].metric}: {results[0].score}")
+            f"Best estimator: {results[0].estimator_name} - "
+            f"{results[0].metric}: {results[0].score}")
 
-        return cls(best_model), ResultGroup(results)
+        return cls(best_estimator), ResultGroup(results)
 
-    def train_model(self) -> 'BaseClassModel':
-        """Loads all training data and trains the model on all data.
-        Typically used as the last step when model tuning is complete.
+    def train_model(self) -> 'ModelData':
```

Comment:
I can't decide if we should go for `train_model` or `train_estimator`. I think it should be the same for `test_X`, `train_X` and `score_X` though